src/waveform/renderers/allshader/waveformrendererrgb.cpp

```diff
@@ -137,63 +137,76 @@ void WaveformRendererRGB::paintGL() {
 
         const float fpos = static_cast<float>(pos);
 
-        // combined left+right
-        float maxLow{};
-        float maxMid{};
-        float maxHigh{};
-        // per channel
-        float maxAll[2]{};
-
+        // Find the max values for low, mid, high and all in the waveform data.
+        // - Max of left and right
+        uchar u8maxLow{};
+        uchar u8maxMid{};
+        uchar u8maxHigh{};
+        // - Per channel
+        uchar u8maxAllChn[2]{};
         for (int chn = 0; chn < 2; chn++) {
             // data is interleaved left / right
             for (int i = visualIndexStart + chn; i < visualIndexStop + chn; i += 2) {
                 const WaveformData& waveformData = data[i];
 
-                const float filteredLow = static_cast<float>(waveformData.filtered.low);
-                const float filteredMid = static_cast<float>(waveformData.filtered.mid);
-                const float filteredHigh = static_cast<float>(waveformData.filtered.high);
-
-                maxLow = math_max(maxLow, filteredLow);
-                maxMid = math_max(maxMid, filteredMid);
-                maxHigh = math_max(maxHigh, filteredHigh);
-
-                const float all = math_pow2(filteredLow) * lowGain +
-                        math_pow2(filteredMid) * midGain +
-                        math_pow2(filteredHigh) * highGain;
-                maxAll[chn] = math_max(maxAll[chn], all);
+                u8maxLow = math_max(u8maxLow, waveformData.filtered.low);
+                u8maxMid = math_max(u8maxMid, waveformData.filtered.mid);
+                u8maxHigh = math_max(u8maxHigh, waveformData.filtered.high);
+                u8maxAllChn[chn] = math_max(u8maxAllChn[chn], waveformData.filtered.all);
             }
         }
 
+        // Cast to float
+        float maxLow = static_cast<float>(u8maxLow);
+        float maxMid = static_cast<float>(u8maxMid);
+        float maxHigh = static_cast<float>(u8maxHigh);
+        float maxAllChn[2]{static_cast<float>(u8maxAllChn[0]), static_cast<float>(u8maxAllChn[1])};
+
+        // Calculate the magnitude of the maxLow, maxMid and maxHigh values
+        const float magnitude = std::sqrt(
+                math_pow2(maxLow) + math_pow2(maxMid) + math_pow2(maxHigh));
+
+        // Apply the gains
         maxLow *= lowGain;
         maxMid *= midGain;
         maxHigh *= highGain;
 
+        // Calculate the magnitude of the gained maxLow, maxMid and maxHigh values
+        const float magnitudeGained = std::sqrt(
```

Comment:
You can use only one std::sqrt() if you build the quotient (factor below) before.

Reply:
Sorry, I understand what you mean here

Reply:
You can make use of the rule:
factor = Sqrt(A)/Sqrt(B) = Sqrt(A/B) 
To get rid of one expensive Sqrt call.

Reply:
ah, well spotted! i'll make the change.

Reply:
Done